src/Management/test/Endpoint.Test/Services/EndpointMiddlewareTest.cs

```diff
@@ -0,0 +1,162 @@
+    {
+        IOptionsMonitor<ServicesEndpointOptions> opts = GetOptionsMonitorFromSettings<ServicesEndpointOptions>();
+        IOptionsMonitor<ManagementEndpointOptions> managementOptions = GetOptionsMonitorFromSettings<ManagementEndpointOptions>();
+
+        var configurationBuilder = new ConfigurationBuilder();
+        configurationBuilder.AddInMemoryCollection(AppSettings).Build();
+
+        IServiceCollection serviceCollection= new ServiceCollection();
+
+        // Add some known services 
+        serviceCollection.AddSingleton<ServicesEndpoint>(); 
+        serviceCollection.AddTransient<ServicesEndpointOptions>();
+        serviceCollection.AddScoped<Startup>();
+
+        var ep = new ServicesEndpoint(opts, serviceCollection, NullLogger<ServicesEndpoint>.Instance);
+        var middle = new ServicesEndpointMiddleware(ep, managementOptions, NullLogger<ServicesEndpointMiddleware>.Instance);
+
+        HttpContext context = CreateRequest("GET", "/beans");
+        await middle.HandleServicesRequestAsync(context);
+        context.Response.Body.Seek(0, SeekOrigin.Begin);
+        var reader = new StreamReader(context.Response.Body, Encoding.UTF8);
+        string json = await reader.ReadLineAsync();
+
+        const string expected =
+           "{\"contexts\":{\"application\":{\"beans\":{\"ServicesEndpoint\":{\"scope\":\"Singleton\",\"type\":\"ServiceType: Steeltoe.Management.Endpoint.Services.ServicesEndpoint Lifetime: Singleton ImplementationType: Steeltoe.Management.Endpoint.Services.ServicesEndpoint\",\"resource\":\"Steeltoe.Management.Endpoint.Services.ServicesEndpoint, Steeltoe.Management.Endpoint, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null\",\"dependencies\":[\"Void .ctor(Microsoft.Extensions.Options.IOptionsMonitor\\u00601[Steeltoe.Management.Endpoint.Services.ServicesEndpointOptions], Microsoft.Extensions.DependencyInjection.IServiceCollection, Microsoft.Extensions.Logging.ILogger\\u00601[Steeltoe.Management.Endpoint.Services.ServicesEndpoint])\"]},\"ServicesEndpointOptions\":{\"scope\":\"Transient\",\"type\":\"ServiceType: Steeltoe.Management.Endpoint.Services.ServicesEndpointOptions Lifetime: Transient ImplementationType: Steeltoe.Management.Endpoint.Services.ServicesEndpointOptions\",\"resource\":\"Steeltoe.Management.Endpoint.Services.ServicesEndpointOptions, Steeltoe.Management.Endpoint, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null\",\"dependencies\":[\"Void .ctor()\"]},\"Startup\":{\"scope\":\"Scoped\",\"type\":\"ServiceType: Steeltoe.Management.Endpoint.Test.Services.Startup Lifetime: Scoped ImplementationType: Steeltoe.Management.Endpoint.Test.Services.Startup\",\"resource\":\"Steeltoe.Management.Endpoint.Test.Services.Startup, Steeltoe.Management.Endpoint.Test, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null\",\"dependencies\":[\"Void .ctor(Microsoft.Extensions.Configuration.IConfiguration)\"]}}}}}";
+        Assert.Equal(expected, json);
```

Comment:
The tools that use this do not have the ability to capture extra data. The sensible thing might be to use implementationType as type since we already have the type as name of the dependency.

Reply:
I think we should leave out "type" when there's an implementation factory. Putting the service name there looks incorrect to me.

Reply:
I'm having second thoughts about my previous comment. What shows in Spring for the following cases?
- `services.AddSingleton<ExampleService>();`
- `services.AddSingleton<IExampleService, ExampleService>();`
- `services.AddSingleton<IExampleService>(sp => new ExampleService());`
- `services.AddSingleton(sp => new ExampleService());`

Reply:
Tracked at https://github.com/SteeltoeOSS/Steeltoe/pull/1224#pullrequestreview-1757640519.

Reply:
Here are some common ways to register dependencies in SpringBoot applications:

Singleton: 
```
@Bean
@Scope("singleton")
public Person personSingleton() {
return new Person();
}
```
produces this in the /actuator/beans:
```
"personSingleton": {
"aliases": [],
"scope": "singleton",
"type": "[com.example.demo.Person](http://com.example.demo.person/)",
"resource": "[com.example.demo.DemoApplication](http://com.example.demo.demoapplication/)",
"dependencies": [
"demoApplication"
]
},
```

Here is "prototype" scope equivalent to "transient" :
```
@Bean
@Scope("prototype")
public Person personScoped() {
return new Person();
}
```

which produces: 
```
"personScoped": {
"aliases": [],
"scope": "prototype",
"type": "[com.example.demo.Person](http://com.example.demo.person/)",
"resource": "[com.example.demo.DemoApplication](http://com.example.demo.demoapplication/)",
"dependencies": []
},
```
Finally when 
Registered Via Interface: 
```
@Bean
@Scope("singleton")
public PersonInterface personInterfaceSingleton() {
return new Person();
}
```
produces: 
```
"personInterfaceSingleton": {
"aliases": [],
"scope": "singleton",
"type": "[com.example.demo.Person](http://com.example.demo.person/)",
"resource": "[com.example.demo.DemoApplication](http://com.example.demo.demoapplication/)",
"dependencies": [
"demoApplication"
]
},
```

Finally there the ability to register new beans via registerBean methods, but I found it hard to get access to that in my sample app. It is not typically done I think unless someone is implementing complex autoconfiguration logic in configuration beans for example. 

I think before we get too involved in implementation, seeing the registered types is a useful feature - just by itself. Neither the mechanisms of registration nor its semantics has exact 1-1 between spring and dotnet and I don't think we should worry about it at this stage.

Reply:
Thanks, this is very helpful!

Reply:
@hananiel  I've made some changes and added tests in https://github.com/SteeltoeOSS/Steeltoe/pull/1224/commits/b521534eceb4005c3ea3ecb15f415e559801af45. Please let me know if it doesn't look right.